test/e2e/helloworld_test.go

```diff
@@ -104,3 +105,109 @@ func TestHelloWorld(t *testing.T) {
 		t.Fatalf("Failed to get Service name from Revision label")
 	}
 }
+
+func TestHelloWorldUserPort(t *testing.T) {
+	var userPort int32
+	userPort = 8888
+	clients := Setup(t)
+
+	//add test case specific name to its own logger
+	logger := logging.GetContextLogger("TestHelloWorld-UserPort")
+
+	newYamlFilename, names, err := CreateConfigAndRouteFromYaml(logger,
+		"helloworld-userport",
+		"../test_images/helloworld-userport/helloworld.yaml",
+		"configuration-example",
+		"route-example",
+		"github.com/knative/serving/test_images/helloworld-userport",
+		namespacePlaceholder,
+	)
+	if err != nil {
+		t.Fatalf("Failed to create configuration and route: %v", err)
+	}
+	defer TearDownByYaml(newYamlFilename, logger)
+	test.CleanupOnInterrupt(func() { TearDownByYaml(newYamlFilename, logger) }, logger)
+
+	logger.Infof("When the Revision can have traffic routed to it, the Route is marked as Ready.")
+	if err := test.WaitForRouteState(clients.ServingClient, names.Route, test.IsRouteReady, "RouteIsReady"); err != nil {
+		t.Fatalf("The Route %s was not marked as Ready to serve traffic: %v", names.Route, err)
+	}
+
+	route, err := clients.ServingClient.Routes.Get(names.Route, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Error fetching Route %s: %v", names.Route, err)
+	}
+	domain := route.Status.Domain
+
+	_, err = pkgTest.WaitForEndpointState(
+		clients.KubeClient,
+		logger,
+		domain,
+		pkgTest.Retrying(pkgTest.MatchesBody(helloWorldExpectedOutput), http.StatusNotFound),
+		"HelloWorldServesText",
+		test.ServingFlags.ResolvableDomain)
+	if err != nil {
+		t.Fatalf("The endpoint for Route %s at domain %s didn't serve the expected text \"%s\": %v", names.Route, domain, helloWorldExpectedOutput, err)
+	}
+
+	logger.Infof("Check user deployment's Ports and Env info.")
+	config, err := clients.ServingClient.Configs.Get(names.Config, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Configuration %s was not updated with the new revision: %v", names.Config, err)
+	}
+	deploymentName :=
+		config.Status.LatestCreatedRevisionName + "-deployment"
```

Comment:
nit: can you pull this out to a getDeployment method in test?

Reply:
sure